System/PathFinder.cpp

```diff
@@ -422,4 +396,90 @@ namespace RTE {
 		}
 		return totalCostOfAdjacentNodes / std::max(static_cast<float>(count), 1.0F);
 	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool PathFinder::UpdateNodeList(const std::unordered_set<int> &nodeList) {
+		std::atomic<bool> anyChange = false;
+
+		// "Hah!" - I hear you say. "This silly person who wrote this code is doing a pointless copy."
```

Comment:
While I appreciate the story and the journey it took me through, I think a less verbose/more succinct comment would be more helpful for future readers.

Reply:
I disagree. Code ought to have some flavor

Reply:
The question of flavour is subjective I guess. I prefer to keep that stuff in commit messages, so imo this is cruft (even if it's fun cruft).

Your other argument is silly, one can easily explain this with fewer words:
`//TODO This copy is necessary because std::for_each with parallel execution doesn't work with std::unordered_set - using it will cause nodes to randomly fail to update. This should be rechecked when the codebase upgrades to cpp20.`

Anyway, this will all be moot if an unordered_set isn't used here.